### ISSUE TITLE
Encryption subkey should be cross-certified

### ISSUE THREAD

**Issue by jrruethe** [romanz/trezor-agent#356]
TLDR; I'll be submitting a PR shortly to fix this.

With the current version (commit `338a075ed500f210a707cb3adf90104602e35c48`), if I generate a key using the Trezor, I get something like this:

```
pub   nistp256 1970-01-01 [SCA]
      39597EBBBC8A6E2B7B8C29929DB331349C28E5AA
uid           [ultimate] test@example.com
sub   nistp256 1970-01-01 [E]
```

Here is the full public key for this test:
```
-----BEGIN PGP PUBLIC KEY BLOCK-----

mFIEAAAAABMIKoZIzj0DAQcCAwQNo6GR8B3I2QYfkEjkRgw5eb4ToH9lrOSEOrHI
LuklOiyXZIWoDc4NTwfAmx55REyZtktdip+fS6YLwGh0YcEBtBB0ZXN0QGV4YW1w
bGUuY29tiIAEExMIABwFAgAAAAACCwkCGwMEFQgJCgQWAgMBAheAAh4BABYJEJ2z
MTScKOWqCxpUUkVaT1ItR1BHbVYBAJl375dN6BOaBjXFPNZ5P18ixRFiU2QIxFs/
wvPNd6cTAP9pYpGngp62F/jIxib7XyRSfKyiVWWPchANsG2D5DdrSLhWBAAAAAAS
CCqGSM49AwEHAgMEoUOrwmz3TFxx7CFKxi/GVb6y5fvsJOHCPSAN/ynijJ2pz/ky
nU1LkURv2YKQL+M6jF4Gxk7McpT2FoY8E+LH1QMBCAeIbQQYEwgACQUCAAAAAAIb
DAAWCRCdszE0nCjlqgsaVFJFWk9SLUdQR5pbAQDHNNbHq+1ozG2Iqu8HuPAd1cWy
xqhW33jILHdRhJVUQgD+MbSxdrTVyIPLBEocCrCyf4jP+waPVeiMBciwK26QNss=
=bfnc
-----END PGP PUBLIC KEY BLOCK-----
```

Using the `hokey lint` tool (from [here](https://hackage.haskell.org/package/hopenpgp-tools), described a bit better [here](https://help.riseup.net/en/security/message-security/openpgp/best-practices)), I get the following output:

```
hokey (hopenpgp-tools) 0.22
Copyright (C) 2012-2019  Clint Adams
hokey comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it under certain conditions.

Key has potential validity: good
Key has fingerprint: 3959 7EBB BC8A 6E2B 7B8C  2992 9DB3 3134 9C28 E5AA
Checking to see if key is OpenPGPv4: V4
Checking to see if key is RSA or DSA (>= 2048-bit): ECDSA 256
Checking user-ID- and user-attribute-related items:
  test@example.com:
    Self-sig hash algorithms: [SHA-256]
    Preferred hash algorithms: [SHA-256, SHA-384, SHA-512]
    Key expiration times: []
    Key usage flags: [[sign-data, certify-keys]]
Checking subkeys:
  one of the subkeys is encryption-capable: True
  fpr: 78AB E1E8 DC68 39AA 9315  93B4 618A 2AAA 69DA AA38
    version: v4
    timestamp: 19700101-000000
    algo/size: ECDH 256
    binding sig hash algorithms: [SHA-256]
    usage flags: [[encrypt-storage, encrypt-communications]]
    embedded cross-cert: False
    cross-cert hash algorithms: [SHA-256]
```

In particular, the second from the last line (`embedded cross-cert: False`) is what this issue is about. 
According to [this page](https://gnupg.org/faq/subkey-cross-certify.html), the best practice is to have subkeys cross-certified using an embedded signature. If I understand correctly, [page 20 of RFC 4880](https://datatracker.ietf.org/doc/html/rfc4880) recommends using the embedded signature on all subkeys:

```
   0x18: Subkey Binding Signature
       This signature is a statement by the top-level signing key that
       indicates that it owns the subkey.  This signature is calculated
       directly on the primary key and subkey, and not on any User ID or
       other packets.  A signature that binds a signing subkey MUST have
       an Embedded Signature subpacket in this binding signature that
       contains a 0x19 signature made by the signing subkey on the
       primary key and subkey.

   0x19: Primary Key Binding Signature
       This signature is a statement by a signing subkey, indicating
       that it is owned by the primary key and subkey.  This signature
       is calculated the same way as a 0x18 signature: directly on the
       primary key and subkey, and not on any User ID or other packets.
```

In the code, I believe the only change needed to correct this is by removing [this](https://github.com/romanz/trezor-agent/blob/338a075ed500f210a707cb3adf90104602e35c48/libagent/gpg/encode.py#L60) `if` statement completely, and generating the embedded signature for the encryption subkey.

As a test, I did exactly that, and regenerated the test key. Results below:

```
hokey (hopenpgp-tools) 0.22
Copyright (C) 2012-2019  Clint Adams
hokey comes with ABSOLUTELY NO WARRANTY. This is free software, and you are welcome to redistribute it under certain conditions.

Key has potential validity: good
Key has fingerprint: 3959 7EBB BC8A 6E2B 7B8C  2992 9DB3 3134 9C28 E5AA
Checking to see if key is OpenPGPv4: V4
Checking to see if key is RSA or DSA (>= 2048-bit): ECDSA 256
Checking user-ID- and user-attribute-related items:
  test@example.com:
    Self-sig hash algorithms: [SHA-256]
    Preferred hash algorithms: [SHA-256, SHA-384, SHA-512]
    Key expiration times: []
    Key usage flags: [[sign-data, certify-keys]]
Checking subkeys:
  one of the subkeys is encryption-capable: True
  fpr: 78AB E1E8 DC68 39AA 9315  93B4 618A 2AAA 69DA AA38
    version: v4
    timestamp: 19700101-000000
    algo/size: ECDH 256
    binding sig hash algorithms: [SHA-256]
    usage flags: [[encrypt-storage, encrypt-communications]]
    embedded cross-cert: True
    cross-cert hash algorithms: [SHA-256]
```

Note how the embedded cross-cert for the encryption subkey is now true.
I will be submitting a pull request for this change.

For completeness, here is the new key after the change

```
-----BEGIN PGP PUBLIC KEY BLOCK-----

mFIEAAAAABMIKoZIzj0DAQcCAwQNo6GR8B3I2QYfkEjkRgw5eb4ToH9lrOSEOrHI
LuklOiyXZIWoDc4NTwfAmx55REyZtktdip+fS6YLwGh0YcEBtBB0ZXN0QGV4YW1w
bGUuY29tiIAEExMIABwFAgAAAAACCwkCGwMEFQgJCgQWAgMBAheAAh4BABYJEJ2z
MTScKOWqCxpUUkVaT1ItR1BHbVYBAJl375dN6BOaBjXFPNZ5P18ixRFiU2QIxFs/
wvPNd6cTAP9pYpGngp62F/jIxib7XyRSfKyiVWWPchANsG2D5DdrSLhWBAAAAAAS
CCqGSM49AwEHAgMEoUOrwmz3TFxx7CFKxi/GVb6y5fvsJOHCPSAN/ynijJ2pz/ky
nU1LkURv2YKQL+M6jF4Gxk7McpT2FoY8E+LH1QMBCAeIzQQYEwgACQUCAAAAAAIb
DAB2CRCdszE0nCjlql8gBBkSCAAGBQIAAAAAAAoJEGGKKqpp2qo44C8BALqFdtos
UZlbLCy0MSjJIqI0RFZjVz6mXBwh7ICaKBIvAPsEx7Mg+FDoS2nx1JK6KxWgFSX7
9vKe6mrK1MS00iJ3xwsaVFJFWk9SLUdQR5pbAQDHNNbHq+1ozG2Iqu8HuPAd1cWy
xqhW33jILHdRhJVUQgD+MbSxdrTVyIPLBEocCrCyf4jP+waPVeiMBciwK26QNss=
=a7te
-----END PGP PUBLIC KEY BLOCK-----
```

And comparing the packets of the keys, to ensure that aside from the cross-certify signature, they keys are the same:

```
$cat before.asc | gpg --list-packets --verbose > before.txt
$cat after.asc | gpg --list-packets --verbose > after.txt
$ diff before.txt after.txt 
31c31
< # off=320 ctb=88 tag=2 hlen=2 plen=109
---
> # off=320 ctb=88 tag=2 hlen=2 plen=205
37a38
> 	subpkt 32 len 94 (signature: v4, class 0x19, algo 18, digest algo 8)
```

**Comment by jrruethe** [comment]
I apologize, while working on a PR for issue #355, I realized that I made a mistake reading the documentation. The cross-certification is intended for a signing subkey to sign the primary key, however encryption subkeys don't have signing capability. The pull request uses the primary key to sign the primary key, which is not correct. I will be closing this.
